lib/spack/spack/package_base.py

```diff
@@ -618,6 +618,12 @@ class PackageBase(WindowsRPath, PackageViewMixin, metaclass=PackageMeta):
     #: Store whether a given Spec source/binary should not be redistributed.
     disable_redistribute: Dict[spack.spec.Spec, DisableRedistribute]
 
+    #: Must be defined as a fallback for old specs that don't have the `build_system` variant
+    legacy_buildsystem: str
```

Comment:
Should we instead call this `default_buildsystem`?  does anything currently mandate that a package has a build system variant?

Reply:
We designed the variant in such a way that it's on all packages, so we kind of mandate that. 

`default_buildsystem` would be a better name indeed, but that means also breaking the previous package API in a few rare cases. We can *add* another name and *deprecate* this one, but I'm not sure it's worth it.

Reply:
Ok two things:

1. Yes, please add and deprecate. I think you could just add a property `legacy_buildsystem()` that returns `default_buildsystem` and document that. I think `default` makes a ton more sense and will be clearer (and worth it) in the long run.
2. If there is a default build system, what does it refer to?  If it is `generic`, should the `generic` build system not just be in core?  Or is it ok for a package not to have a build system?  Basically, I think there is still an issue if there is a property that refers to something that *has* to exist in a repo.  I think a v2 package API package should be able to function in a repo that has no build systems -- is that possible currently?  Do you agree?

Reply:
Ok, let me check then if we need to do anything wrt the auto-generated adapter. There are other "legacy" properties that are used in there, if they're present.

Reply:
@alalazo I like that plan -- let's do that so that core has a minimally usable package implementation with a build system, and so that the `build_system` part of the API is not spread across core and repo